views.go

```diff
@@ -619,20 +626,28 @@ func (lk *logKeeper) viewBucketLogs(r *http.Request, buildID string, testID stri
 	}()
 	wg.Wait()
 
+	if pail.IsKeyNotFoundError(buildErr) {
+		if shouldFallBack {
+			if testID == "" {
+				return lk.viewAllDBLogs(r, buildID)
+			} else {
+				return lk.viewDBTestLogs(r, buildID, testID)
+			}
+		} else {
+			return nil, &apiError{Err: "finding build", code: http.StatusNotFound}
+		}
+	}
```

Comment:
could it be advantageous to trigger the fallback as soon as the build goroutine returns? 
This way we wouldn't need to wait on the slowest of the requests.

Reply:
I moved the build fetch into the main thread because that seemed clearer/cleaner than having separate goroutine wait groups.

Reply:
Nice!
Could it make sense to pass a derived context to the other requests and cancel it when in the `pail.IsKeyNotFoundError(buildErr)` case?

Reply:
I hope I did this right.